test/0.8.25/vaults/staking-vault/StakingVaultTest.t.sol

```diff
@@ -0,0 +1,513 @@
+        function() internal[] memory transitions = new function() internal[](1);
+        transitions[0] = transitionRandomReceiveReward;
+        return transitions;
+    }
+
+    function vaultHubTransitions() internal pure returns (function() internal[] memory) {
+        function() internal[] memory transitions = new function() internal[](3);
+        transitions[0] = transitionRandomMintShares;
+        transitions[1] = transitionRandomReport;
+        transitions[2] = transitionRandomRebalance;
+        return transitions;
+    }
+
+    function mergeTransitions(
+        function() internal[] memory _transitionsA,
+        function() internal[] memory _transitionsB
+    ) internal pure returns (function() internal[] memory) {
+        function() internal[] memory result = new function() internal[](_transitionsA.length + _transitionsB.length);
+        for (uint256 txIdx = 0; txIdx < _transitionsA.length; txIdx++) {
+            result[txIdx] = _transitionsA[txIdx];
+        }
+        for (uint256 txIdx = 0; txIdx < _transitionsB.length; txIdx++) {
+            result[_transitionsA.length + txIdx] = _transitionsB[txIdx];
+        }
+        return result;
+    }
+
+    function getAvailableTransitions(
+        bool _hasValidator,
+        bool _isConnectedToHub
+    ) internal pure returns (function() internal[] memory) {
+        if (_hasValidator && _isConnectedToHub) {
+            return mergeTransitions(baseTransitions(), mergeTransitions(validatorTransitions(), vaultHubTransitions()));
+        } else if (_hasValidator && !_isConnectedToHub) {
+            return mergeTransitions(validatorTransitions(), baseTransitions());
+        } else if (!_hasValidator && _isConnectedToHub) {
+            return mergeTransitions(baseTransitions(), vaultHubTransitions());
+        } else {
+            return baseTransitions();
+        }
+    }
+
+    function transitionRandomUserDeposit() internal {
+        console2.log("------Deposit by random user------");
+
+        uint256 amount = rnd.randAmountD18();
+        console2.log("Deposit by random user %d", amount);
+
+        address randomUser = rnd.randAddress();
+        deal(randomUser, amount);
+        if (amount == 0) {
+            vm.expectRevert(abi.encodeWithSelector(ZeroArgument.selector, "msg.value"));
+        }
+        vm.prank(randomUser);
+        payable(address(stakingVaultProxy)).transfer(amount);
+        vm.stopPrank();
+        randomUserDeposits += amount;
+    }
+
+    function transitionRandomFund() internal {
+        console2.log("------Fund vault------");
+
+        uint256 amount = rnd.randAmountD18();
+        console2.log("Fund vault %d", amount);
+
+        deal(owner, amount);
+        if (amount == 0) {
+            vm.expectRevert(abi.encodeWithSelector(ZeroArgument.selector, "msg.value"));
+        }
+        vm.startPrank(owner);
+        stakingVaultProxy.fund{value: amount}();
+        vm.stopPrank();
+        deposits += amount;
```

Comment:
Maybe add asserts that the state changed? here, valuation must increase and inOutDelta